src/main/java/jenkins/scm/api/SCMFileSystem.java

```diff
@@ -388,6 +441,67 @@ public static boolean supports(@NonNull SCMSource source) {
          */
         public abstract boolean supports(SCMSource source);
 
+        /**
+         * Checks if this {@link Builder} supports the supplied {@link SCMDescriptor}.
+         *
+         * @param descriptor the {@link SCMDescriptor}
+         * @return the return value of {@link #supportsDescriptor(SCMDescriptor)} if implemented, otherwise, it
+         * will return true if one of the following is true: the {@link SCM} for the descriptor is the enclosing
+         * class of this builder class, the builder and {@link SCM} are in the same package, or the builder is in
+         * a child package of the {@link SCM}.
+         * @since 2.3.0
+         */
+        public final boolean supports(SCMDescriptor<?> descriptor) {
+            if (Util.isOverridden(SCMFileSystem.Builder.class, getClass(), "supportsDescriptor", SCMDescriptor.class)) {
```

Comment:
Weird. Just make the method non`abstract` and provide the `else` clause as the default implementation. Why is this overcomplicated?

Reply:
@stephenc wanted it this way. =)